ir/base.def

```diff
@@ -253,4 +254,21 @@ interface IInstance {
     virtual Type getType() const = 0;
 }
 
+/// An argument to a function call (or constructor call)
+/// Arguments may have optional names
+class Argument {
+    /// If an argument has no name the name.name is nullptr.
+    optional ID name;
+    Expression expression;
```

Comment:
Arguably the wrong decision was made in the original design, when we have not used an Argument class (which would have contained just one expression). We are only fixing this now.

However, if you look at the code everywhere, accessing IR fields directly by name is the standard procedure in this codebase. All IR fields are public, and are used as public. There are no accessor methods whatsoever.

Adding `operator()` would not reduce the number of changes.

Since I am not using any fancy IDE, refactoring expression to arg_value would have to be done again manually - probably a few hours of work. (Using Java that could have been done in 2 seconds...).

Moreover, the second part of this change that is upcoming will probably be even harder to make - match arguments to parameters; for the current change the type system pointed out most places that needed to be changed, but for the next one there will be no help. I expect that change to take at least a couple of days of work.